src/components/Common/SearchBarDropdown.tsx

```diff
@@ -0,0 +1,207 @@
+    []
+  );
+
+  const fetchUsers = async (keywords: string) => {
+    setLoading(true);
+    if (keywords.length === 0) {
+      return;
+    }
+
+    try {
+      const token = await getAccessTokenSilently();
+      const result = await axios.get(
+        `http://localhost:3001/api/users/searchUsers`,
+        {
+          params: {
+            keyword: keywords,
+          },
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+
+      setSearchOptions(result.data as SelectedUser[]);
+
+      if (keywords.trim() === "") {
+        setSearchOptions([]);
+      }
+    } catch (error) {
+      console.error("Failed to fetch users:", error);
+      setSearchOptions([]);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    debouncedFetch(keywords);
+  }, [keywords]);
+
+  const handleInputChange = (newInputValue: string) => {
+    setKeywords(newInputValue);
+
+    if (newInputValue.trim() === "") {
+      setSearchOptions([]);
+    }
+  };
+
+  const handleSelect = (selectedUsername: string) => {
+    setKeywords("");
+    setSearchOptions([]);
+    inputRef.current?.blur();
+    navigate(`/${selectedUsername}`);
+  };
+
+  const handleClear = () => {
+    setKeywords("");
+    setSearchOptions([]);
+    setFocusSearchBar(false);
+  };
+
+  return (
+    <Box sx={styles.box}>
+      <Autocomplete
+        disablePortal
+        fullWidth
+        filterOptions={(x) => x}
+        getOptionLabel={() => ""}
+        forcePopupIcon={false}
+        id="search"
+        value={null}
+        inputValue={keywords}
+        ListboxProps={{ style: styles.listBox }}
+        loading={!keywords && loading}
+        loadingText="Start typing to search..."
+        onChange={(_, value) => value && handleSelect(value.username)}
+        onBlur={handleClear}
+        onInputChange={(_, newInputValue) => handleInputChange(newInputValue)}
+        onFocus={() => setFocusSearchBar(true)}
+        open={focusSearchBar}
+        openOnFocus
+        options={searchOptions}
+        noOptionsText={"No user found"}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            fullWidth
+            variant="outlined"
+            inputRef={inputRef}
+            InputProps={{
+              ...params.InputProps,
+              startAdornment: (
+                <InputAdornment position="start">
+                  <IconButton
+                    disabled
+                    sx={
+                      focusSearchBar
+                        ? styles.searchIconFocused
+                        : styles.searchIcon
+                    }
+                  >
+                    <SearchRoundedIcon />
+                  </IconButton>
+                </InputAdornment>
+              ),
+              endAdornment: keywords && (
+                <InputAdornment position="end">
+                  <IconButton onClick={() => setKeywords("")}>
```

Comment:
The search options don't get cleared here, so when you click the clear button the dropdown options are still available

Reply:
Good point, I'll just use `handleClear` here instead